### ISSUE TITLE
Can't compile without compile key

### ISSUE THREAD

**Issue by izacsc** [nginformatica/advpl-testsuite#3]
I cloned the repository and tried to compile and wasn't able to.

The following compiler directives in [testsuite.ch](https://github.com/nginformatica/advpl-testsuite/blob/master/include/testsuite.ch) creates functions instead of user functions:

```
#xcommand Before TestSuite <cSuite> ;
    => ;
    Function ___TestSuite_<cSuite>____Before()

#xcommand Feature <cFeat> ;
    TestSuite <cSuite> ;
    => ;
    Function ___TestSuite_<cSuite>____Feat_<cFeat>()

#xcommand CompileTestSuite <cSuite> ;
    => ;
    Function ___TestSuite_<cSuite>____New( cName, cDescription ) ;;
        _Super:New( cName, cDescription ) ;;
        Return Self ;;
    Function U_<cSuite> ;;
        Local oTester := TestSuite_<cSuite>():New( __NAME__, __DESC__ ) ;;
        oTester:lVerbose := __VERBOSE__ ;;
        oTester:Run( oTester ) ;;
        Return 0
```

In [fluentexpr.prw](https://github.com/nginformatica/advpl-testsuite/blob/master/src/fluentexpr.prw), ReadFileContents is a Function. 

The [example.prw](https://github.com/nginformatica/advpl-testsuite/blob/master/examples/example.prw) fail to compile with:
```
EXAMPLE.PRW(4)   C2003 Syntax Error
```

**Comment by endersonmaia** [comment]
I don't know if it's the most elegant solution, but maybe you should send the function `ReadFileContents` to `advpl-testsuite/include/testsuite.ch` as a `STATIC FUNCTION`

HTH

**Comment by haskellcamargo** [comment]
`example.prw` fails to compile because `Environment X Y` is now `Enabled Environment X Y`. We have an open PR to fix that. Changing it in your file will also fix.
For now, we really have some limitations with user functions. Class methods are indeed only functions with a specific pattern (`___CLASSNAME____METHODNAME`), so I'm not sure if they compile without a compilation key.

**Comment by haskellcamargo** [comment]
`ReadFileContents` was previously a `Static Function`, but I've changed to avoid repeating the call inside `testsuite.prw`. I believe the right way here would be, so, using a `StaticCall` from `testsuite.prw`.

**Comment by haskellcamargo** [comment]
@endersonmaia cool! having it in the preprocessor would also be a good approach!

**Comment by endersonmaia** [comment]
@haskellcamargo I'd add some watchdogs`#define` in the `.ch` too

ex.:

```
#ifndef _TESTSUITE_CH
#define _TESTSUITE_CH

...

#endif //_TESTSUITE_CH 
```

maybe I'll send a PR, but for now it's here 😄

**Comment by izacsc** [comment]
@haskellcamargo really interesting. I changed to enable environment and it actually compiles. Crazy stuff!

I had commented the TestSuite block to compile and it emits the ```Erro in example.prw EXAMPLE.PRW(12)   Regular functions are not allowed in code. Use USER FUNCTION or STATIC FUNCTION at line 12```

When I remove the comments ( after changing to enable environment) it works. Thanks!

**Comment by haskellcamargo** [comment]
Fixed by @endersonmaia via #7
